git_flow.rst

```diff
@@ -0,0 +1,185 @@
+.. image:: https://cdn.rawgit.com/LABSN/expyfun/master/doc/git_flow.svg
+
+
+Users
+^^^^^
+Users who want access to the source code typically do something like this::
+
+    $ git clone git://github.com/LABSN/expyfun.git
+    $ cd expyfun
+    $ python setup.py install
+
+Perhaps without realizing it, these users have set up a structure so that
+their local repository knows about a remote repository named "origin" that
+points to `<http://github.com/LABSN/expyfun.git>`_, and they have checked out
+the branch "master" from that repository. This can be seen by doing::
+
+    $ git remote -v
+    origin	git://github.com/LABSN/expyfun.git (fetch)
+    origin	git://github.com/LABSN/expyfun.git (push)
+    $ git branch
+    * master
+
+This setup is convenient, because to update their local repo they can just do::
+
+    $ git pull
+
+Under the hood, this is doing something more like this::
+
+    $ git pull origin/master
+
+And thus they can keep code up to date with the remote repo easily.
+
+
+Developers
+^^^^^^^^^^
+Developers typically use a slightly different setup. The first step is to go
+onto GitHub.com and fork the repository so that changes can be sandboxed.
+If the developer had account :bash:`rkmaddox` It is more standard to
+have :bash:`origin` point to their own fork. The easiest way to do this is::
+
+    $ git clone git@github.com:/rkmaddox/expyfun.git
+    $ cd expyfun
+    $ git remote add upstream git://github.com/LABSN/expyfun.git
+
+Now they are set up with the standard :bash:`origin`/:bash:`upstream` flow
+as::
+
+    $ git remote -v
+    origin	git@github.com:/rkmaddox/expyfun.git (fetch)
+    origin	git@github.com:/rkmaddox/expyfun.git (push)
+    upstream	git://github.com/LABSN/expyfun.git (fetch)
+    upstream	git://github.com/LABSN/expyfun.git (push)
+    $ git branch
+    * master
+
+Now to make a change, the flow would be something like this (to e.g., add
+some trivial file)::
+
+    $ git branch fix_branch
+    $ git branch
+      fix_branch
+    * master
+    $ git checkout fix_branch
+    $ git branch
+    * fix_branch
+      master
+    $ touch foo.txt
+    $ git add foo.txt
+    $ git commit -am 'FIX: Add missing foo file'
+    $ git push origin fix_branch
+
+This creates a new branch called :bash:`fix_branch` on the local machine,
+checks out that branch, adds a file, commits the change to the branch, and
+then pushes the branch to the :bash:`origin` repo. The user could then
+navigate to `<http://github.com/LABSN/expyfun/>`_ and they would find a nice
+**Pull Request** button available to open a pull request.
+
+Maintainers_ would then typically comment on the pull request and ask for
+some changes. For example, maybe the user forgot to also add the necessary
+:bash:`bar` file. The user would then do::
+
+    $ git branch
+    * fix_branch
+      master
+    $ touch bar.txt
+    $ git add bar.txt
+    $ git commit -am 'FIX: Add missing file'
+    $ git push origin fix_branch
+
+After this set of commands, the 
```

Comment:
I'm guessing you wanted something like this (but feel free to modify):

"After this set of commands, the pull request is automatically updated to reflect this new addition. The cycle of commenting on and updating the continues until the Maintainers_ are satisfied with the changes. They will then merge the pull request to incorporate the Developer_'s proposed changes into the GitHub repo."